app/controllers/questions_controller.rb

```diff
@@ -19,9 +23,111 @@ def show
       gon.solution = solution
     end
 
+    if !(current_user.show_solutions)
+      if @room.answered(@question.id,@room.id)
+        @number_of_question = @room.question_count - @room.question_count_not_visited(@room.id) + 1
+      else
+        @number_of_question = @room.question_count - @room.question_count_not_visited(@room.id)
+      end
+    else
+      @number_of_question = @room.learning_objects.where("learning_object_id < ?",@question.id).count + 1
+    end
+
+    @feedbacks = @question.feedbacks.includes(:user)
     @feedbacks = @question.feedbacks.visible.includes(:user)
   end
 
+  def eval_question
+    weight_solved = 5
+    weight_failed = 2
+    weight_dont_now = 1
+    @setup = Setup.take
+    @week = @setup.weeks.find_by_number(params[:week_number])
+
+    room = Room.find(params[:room_number])
+
+    lo_class = Object.const_get params[:type]
+    lo = lo_class.find(params[:id])
+
+    @results = UserToLoRelation.get_results_room(current_user.id,@week.id,params[:room_number], room.number_of_try)
+    result = @results.find {|r| r["result_id"] == lo.id.to_s}
+    unless result.nil?
+      solved = result['solved']
+      failed = result['failed']
+      donotnow = result['donotnow']
+    end
+    room = Room.find(params[:room_number])
+
+    if params[:commit] == 'send_answer'
+      result = lo.right_answer? params[:answer], @solution
+    end
+
+    difficulty = lo.difficulty.to_s
+    dif_value = 0.0
+    importance = lo.importance.to_s
+    imp_value = 0.0
+
+    if difficulty.to_s == "trivial"
+      dif_value = 0.01
+    elsif difficulty.to_s == "easy"
+      dif_value = 0.25
+    elsif difficulty.to_s == "medium"
+      dif_value = 0.5
+    elsif difficulty.to_s == "hard"
+      dif_value = 0.75
+    elsif difficulty.to_s == "impossible"
+      dif_value = 1
+    else
+      dif_value = 0.5
+    end
```

Comment:
Podla mna je este menej ukecane a udrziavatelnejsie pouzit Hash, napr.:
```
DIFFICULTY_VALUE = {
     trivial: 0.01,
     easy: 0.25,
     medium: 0.5,
     ...
}
```
A potom:
`dif_value = DIFFICULTY_VALUE[difficulty.to_sym]`

Hlavne by tie hodnoty nemali byt zadefinovane rovno v controlleri (bud nejaky konfigurak `*.yml` alebo nejaka libka, ktora toto ucelene riesi) a idealne by to mala robit metoda modelu Question/LearningObject - pytas sa learning objectu `lo` "daj mi svoju obtiaznost" a ten ti uz rovno vrati to ciselko.

Tiez by bolo dobre pomenovavat premenne konkretnejsie. Difficulty value je prilis vseobecne - je to IRT hodnota alebo nieco take?